OpenRA.Game/Map/CellLayer.cs

```diff
@@ -45,6 +45,19 @@ public void CopyValuesFrom(CellLayer<T> anotherLayer)
 			Array.Copy(anotherLayer.entries, entries, entries.Length);
 		}
 
+		public static CellLayer<T> CreatePrototype(Func<MPos, T> creator, Size size, TileShape tileShape)
```

Comment:
I'm thinking  more descriptive name for ```creator``` would be nice. Maybe ```initialCellValueFactory``` or something along those lines.